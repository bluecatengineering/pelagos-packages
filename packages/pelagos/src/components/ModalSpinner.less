@import '../../less/modal';

.ModalSpinner {
	@modal-backdrop();
}
