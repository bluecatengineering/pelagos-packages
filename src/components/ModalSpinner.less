@import '../../less/colors';

.ModalSpinner {
	position: fixed;
	top: 0;
	left: 0;
	bottom: 0;
	right: 0;
	background-color: @modal-bg;
	z-index: 5700;
}
