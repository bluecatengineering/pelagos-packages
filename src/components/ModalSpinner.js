import React from 'react';
import PropTypes from 'prop-types';

import Spinner from './Spinner';
import './ModalSpinner.less';

/** A modal spinner. */
const ModalSpinner = ({...props}) => (
	<div {...props} className="ModalSpinner">
		<Spinner size="large" />
	</div>
);

ModalSpinner.propTypes = {
	/** The component id. */
	id: PropTypes.string,
};

export default ModalSpinner;
