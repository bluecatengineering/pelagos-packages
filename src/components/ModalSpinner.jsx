import React from 'react';
import PropTypes from 'prop-types';

import Spinner from './Spinner';
import './ModalSpinner.less';

const ModalSpinner = ({id, componentId}) => (
	<div id={id} data-bcn-id={componentId} className="ModalSpinner">
		<Spinner size="large" />
	</div>
);

ModalSpinner.propTypes = {
	id: PropTypes.string,
	componentId: PropTypes.string,
};

export default ModalSpinner;
