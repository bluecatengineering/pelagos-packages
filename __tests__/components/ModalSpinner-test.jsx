import React from 'react';
import {shallow} from 'enzyme';

import ModalSpinner from '../../src/components/ModalSpinner';

jest.unmock('../../src/components/ModalSpinner');

describe('ModalSpinner', () => {
	describe('rendering', () => {
		it('renders expected elements', () => {
			const wrapper = shallow(<ModalSpinner id="test" componentId="test-component" />);
			expect(wrapper.getElement()).toMatchSnapshot();
		});
	});
});
