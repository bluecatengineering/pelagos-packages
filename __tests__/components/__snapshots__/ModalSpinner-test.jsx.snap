// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalSpinner rendering renders expected elements 1`] = `
<div
  className="ModalSpinner"
  data-bcn-id="test-component"
  id="test"
>
  <Spinner
    size="large"
  />
</div>
`;
